pages/grid.js

```diff
@@ -0,0 +1,109 @@
+import { flag } from "country-emoji";
+import { motion } from "framer-motion";
+import Link from "next/link";
+import React, { useEffect, useMemo, useState } from "react";
+
+import Controls from "../components/Controls";
+import { Flag } from "../components/Flag";
+import { Intro } from "../components/Intro";
+import Legend from "../components/Legend";
+import MessageUnsupportedBrowser from "../components/MessageUnsupportedBrowser";
+import { Methodology } from "../components/Methodology";
+import { getData } from "../data";
+import { useFlagDimensions } from "../hooks/useFlagDimensions";
+import { nth } from "../utils";
+
+function Grid() {
+  const [isSamsungBrowser, setIsSamsungBrowser] = useState(false);
+  const [year, setYear] = useState(2024);
+  const [orderAlphabetical, setOrderAlphabetical] = useState(true);
+  const [orderRanking, setOrderRanking] = useState(true);
+  const [sortDict, setSortDict] = useState({
+    type: "ranking",
+    ascending: true,
+  });
+  const [flagsContainer, flagDimensions] = useFlagDimensions();
+
+  const data = useMemo(
+    () => getData({ years: [year], sortingParams: sortDict }),
+    [year, sortDict]
+  );
+
+  useEffect(() => {
+    if (navigator && navigator.userAgent.match(/SamsungBrowser/i)) {
+      setIsSamsungBrowser(true);
+    }
+  }, []);
+
+  const countries = data.map((d) => d.country);
+
+  return (
+    <>
+      {!isSamsungBrowser ? (
+        <>
+          {/* <Intro /> */}
+          <div id="flags" className="py-4">
```

Comment:
I'm thinking we could extract the grid to its own component (for the same reasons as my comment on <Header>, keep the bits that are the same re-usable), wdyt? 🤔